src/components/button/Button.tsx

```diff
@@ -190,6 +193,11 @@ export const Button = React.forwardRef<any, ButtonProps>(
               <ThemeIcon fill={textColor} {...rightIcon} />
             </View>
           )}
+          {!rightIcon && loading && (
+            <View style={rightStyling}>
+              <ActivityIndicator size="small" color={styleText.color} />
+            </View>
+          )}
```

Comment:
Does it make sense that a button with an icon on the right side can't have a loading state? 🤔

Reply:
The ButtonIconProps used for rightButton includes loading. So currently a button which has an icon can show a spinner instead of the icon when loading. So this will allow a button which doesn't have an icon to show the loader as well. 
That's sort of what I hackishly tried to get around here: https://github.com/AtB-AS/mittatb-app/pull/3924/files#diff-878b44656b464aeb7303c00c30a3f811dee1b5609a0f2dd6d7449540ad09b804R228-R235
On the other hand we could maybe remove loading from ButtonIconProps, and either show ThemeIcon or ActivityIndicator a few lines above instead: https://github.com/AtB-AS/mittatb-app/pull/4076/files/c99a60d6cba4c642d1dcfb57837dd188638a43be#diff-dbb0a2706cf9edd17c0ff507bbe8bc37e950d6e7cdb77116e7ca9d5d0c29aa72R193

Reply:
Yeah, I think it actually can be removed from `ButtonIconProps` and `ThemeIconProps`, and that we just use the `loading` parameter for the `Button` instead :+1:

Reply:
Made changes accordingly now 👩🏼‍🚀